frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/OptionListEditor.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { createRef, useState } from 'react';
+import type { IGenericEditComponent } from '@altinn/ux-editor/components/config/componentConfig';
+import type { SelectionComponentType } from '@altinn/ux-editor/types/FormComponent';
+import type { Option } from 'app-shared/types/Option';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCodeListEditor,
+  StudioModal,
+  StudioSpinner,
+  StudioErrorMessage,
+} from '@studio/components';
+import { TableIcon } from '@studio/icons';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from '../../../../../hooks/mutations/useUpdateOptionListMutation';
+import { useOptionListsQuery } from '../../../../../hooks/queries/useOptionListsQuery';
+import { useOptionListEditorTexts } from './hooks/useOptionListEditorTexts';
+import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import classes from './OptionListEditor.module.css';
+
+type OptionListEditorProps = Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+export function OptionListEditor({ component }: OptionListEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: optionsListMap, status } = useOptionListsQuery(org, app);
+
+  switch (status) {
+    case 'pending':
+      return (
+        <StudioSpinner spinnerTitle={t('ux_editor.modal_properties_code_list_spinner_title')} />
+      );
+    case 'error':
+      return (
+        <StudioErrorMessage>{t('ux_editor.modal_properties_error_message')}</StudioErrorMessage>
```

Comment:
Maybe we should adapt this text a bit? It is difficult to see exactly _what error occurred during loading_ here 😅 
Maybe we should get @Ildest on the field for this one 💪 

![Skjermbilde 2024-11-06 kl  12 42 01](https://github.com/user-attachments/assets/1c0b94cf-628d-4913-9e54-6a1b68dc10a4)

Reply:
To clearify: with this comment I mean that it is difficult to see what the user was suppose to see if there was not an error 😊

Reply:
Suggestion for amended message, after huddle with @Konrad-Simso: "Kunne ikke hente kodelisten. Prøv igjen."

We landed on a simple "Prøv igjen" without specifying what to try, as this is an error that does not occur very often, and the users who experience it will most likely know why they get it.

GitHub says: "You cannot use the suggestion function on an outdated comment", Konrad, so if you can implement the suggestion in code, that is grand :-).

Reply:
Will do

Reply:
Would there be an idea to include "Kunne ikke hente innholdet i kodelisten. Redigering er ikke tilgjengelig". To specify what would be available if the error was not there? 🙈